modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceCatalogLocalServiceImpl.java

```diff
@@ -145,13 +145,15 @@ public CommerceCatalog deleteCommerceCatalog(
 
 		validate(commerceCatalog);
 
+		long groupId = commerceCatalog.getGroupId();
+
 		// Commerce catalog
 
 		commerceCatalogPersistence.remove(commerceCatalog);
 
 		// Group
 
-		groupLocalService.deleteGroup(commerceCatalog.getGroupId());
+		groupLocalService.deleteGroup(groupId);
```

Comment:
@riccardo-alberti I think itd be more clear if we just moved
```
// Group

groupLocalService.deleteGroup(commerceCatalog.getGroupId());
```
before
```
// Commerce catalog

commerceCatalogPersistence.remove(commerceCatalog);
```

if that solves NoSuchGroup exception

Reply:
@dsitu if the persistence is removed after the group is deleted the model listener on the catalog will fail with NoSuchGroupException